pkg/api/types.go

```diff
@@ -2128,3 +2128,35 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// A ThirdPartyObject is a generic representation of a resource, it is used by add-ons and plugins to add new resource
+// types to the API.  It consists of one or more Versions of the api.
+type ThirdPartyObject struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	Description string `json:"description,omitempty" description:"The description of this object"`
+
+	Versions []Version `json:"versions,omitempty" description:"The versions for this third party object"`
```

Comment:
We need to specify the Kind. Assuming that we never support subresources for such extensions, we could automatically derive the resource collection by lowercasing and pluralifying.

Why do we need a list here? I don't imagine supporting conversion for these types.